homeassistant/components/wilight/light.py

```diff
@@ -0,0 +1,223 @@
+"""Support for WiLight lights."""
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from . import WiLightDevice
+from .const import (
+    DOMAIN,
+    ITEM_LIGHT,
+    LIGHT_COLOR,
+    LIGHT_DIMMER,
+    LIGHT_NONE,
+    LIGHT_ON_OFF,
+    SUPPORT_NONE,
+)
+
+
+def entities_from_discovered_wilight(hass, api_device):
+    """Parse configuration and add WiLight light entities."""
+    entities = []
+    for item in api_device.items:
+        if item["type"] != ITEM_LIGHT:
+            continue
+        if item["sub_type"] == LIGHT_NONE:
+            continue
+        index = item["index"]
+        item_name = item["name"]
+        if item["sub_type"] == LIGHT_ON_OFF:
+            entity = WiLightLightOnOff(api_device, index, item_name)
+        elif item["sub_type"] == LIGHT_DIMMER:
+            entity = WiLightLightDimmer(api_device, index, item_name)
+        elif item["sub_type"] == LIGHT_COLOR:
+            entity = WiLightLightColor(api_device, index, item_name)
+        else:
+            continue
```

Comment:
The for loop will continue by default when reaching the end.
```suggestion
```

Reply:
But I want to add entities except in the case else...

entities.append(entity)

Reply:
Right.